### ISSUE TITLE
A missing image in bxslider prevents bx-loading div from hiding

### ISSUE THREAD

**Issue by djmtype** [stevenwanderski/bxslider-4#560]
If an image being called is missing in one of my slides, this causes the bx-loading div to remain present, in front of the slides and it never goes away. 

There are no JS errors. The carousel functions fine, if I remove `bx-loading` div via web inspector. 

Using uncompressed version bxslider 4.1.2

How I'm calling bxslider…
```
jQuery('.col-md-12 .carousel-horizontal ul').bxSlider({
  mode: 'horizontal',
  slideMargin: 10,
  minSlides: 1,
  maxSlides: 6,
  pager: false,
  infiniteLoop: false,
  responsive: true,
  touchEnabled: true,
  moveSlides: 0,
  slideWidth: 212
});
```

**Comment by Tidal-Wave** [comment]
We are looking to address the issue of the slider not removing the loading if an image is missing in a upcoming update. The PR issued above doesn't address the issue so it has been closed.

**Comment by abhogal-kpmg** [comment]
i managed to write this hack-fix to get it to force the callback method (that contains onSliderLoad) - 

im using bootstrap +angular for a content slider, i debugged the bxslider code and realised some methods were not being called back

the counter messes up on broken images so i just simplified it (have not tested for anything else it may break)

it should replace the loadElements method on line 271 in jquery.bxslider.js

		var loadElements = function (selector, callback) {

		    var total = selector.find('img, iframe').length;

		    if (total == 0) {
		        console.log("LoadElements3");

		        callback();
		        console.log("LoadElements4");

				return;
		    }

		    var count = 0;
			selector.find('img, iframe').each(function () {

			    if (++count === total) {

			        callback();

			    } else {
			        if (this.complete) $(this).load();

			    }
			});
		}

**Comment by Tidal-Wave** [comment]
Fixed in 4.2.2

**Comment by creativename** [comment]
I seem to still be having this issue when using 4.2.2. The loading div never goes away for some of my sliders.

**Comment by Tidal-Wave** [comment]
Due to a small error from a unrelated patch. Please try to be recent 4.2.3

**Comment by creativename** [comment]
Thanks. This seems to fix it. I was updating via Bower and it only pulled version 4.2.2.

**Comment by eranitach** [comment]
Thanks!! it works like a charm!

**Comment by Tidal-Wave** [comment]
Sorry about the bower issue, the 4.2.3 tag was not in place.

**Comment by jpalo** [comment]
Still seeing this on first reload after cache has been cleared on IE11.

**Comment by harveyelsisura** [comment]
> We just tried this solution and although it works sometimes, the bx-loading gets stuck loading sometimes. If we do a display:none the images underneath - although loaded - won't allow the slider to work. Any thoughts? Thanks!!!!!

Hello, Did you find any solutions?

**Comment by eranitach** [comment]
> > We just tried this solution and although it works sometimes, the bx-loading gets stuck loading sometimes. If we do a display:none the images underneath - although loaded - won't allow the slider to work. Any thoughts? Thanks!!!!!
> 
> Hello, Did you find any solutions?

Please provide a working plunker or something like that, to allow us to see the code you're using.

**Comment by kenwin007** [comment]
Is 4.2.3 version available? I did not find this. And the loading div problem is solved?

**Comment by jordanrigo** [comment]
I still encounter this issue. Do you have a fix for this?

**Comment by hirushacooray** [comment]
I'm also still facing this issue.

**Comment by robp** [comment]
I ran into this issue recently. Depending on how your slides are being generated the image HTML output may be using lazy loading:

```
<img ... loading="lazy">
```

The carousel won't remove the loading spinner until all images are loaded, and if you're using lazy loading then the images won't load until the carousel is scrolled to reveal (or almost reveal) them on screen. Remove lazy loading from your images and that may fix your issue.

In our case we just needed to add `'loading' => false` to `$attr` for `get_the_post_thumbnail()`.
